core/src/actors/paths.rs

```diff
@@ -306,29 +302,33 @@ impl ActorPath {
 
     /// Same as [tell](ActorPath::tell), but serialises eagerly into a Pooled buffer (pre-allocated and bounded)
     pub fn tell_serialised<CD, B>(&self, m: B, from: &CD) -> Result<(), SerError>
-    where
-        CD: ComponentDefinition + Sized + 'static,
-        B: Serialisable + 'static,
+        where
+            CD: ComponentDefinition + Sized + 'static,
+            B: Serialisable + 'static,
     {
         if self.protocol() == Transport::LOCAL {
             // No need to serialize!
             self.tell(m, from);
             return Ok(());
         }
-        //let mut buf = encode_buffer.deref_mut();
-        let mut buf_ref = from.ctx().get_buffer().borrow_mut();
-        let buf = &mut EncodeBuffer::get_buffer_encoder(buf_ref.deref_mut());
-
-        let chunk_lease =
-            crate::serialisation::ser_helpers::serialise_msg(&from.actor_path(), &self, &m, buf)?;
-
-        let env = DispatchEnvelope::Msg {
-            src: from.path_resolvable(),
-            dst: self.clone(),
-            msg: DispatchData::Serialised((chunk_lease, m.ser_id())),
-        };
-        from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env));
-        Ok(())
+        // Scope the buffer check so we can safely initialise it if we need to
+        {
+            if let Some(ref mut buffer) = (*from.ctx().get_buffer().borrow_mut()).as_mut() {
+                let mut buf = buffer.get_buffer_encoder();
+                let chunk_lease = crate::serialisation::ser_helpers::serialise_msg(&from.actor_path(), &self, &m, &mut buf)?;
+
+                let env = DispatchEnvelope::Msg {
+                    src: from.path_resolvable(),
+                    dst: self.clone(),
+                    msg: DispatchData::Serialised((chunk_lease, m.ser_id())),
+                };
+                from.dispatcher_ref().enqueue(MsgEnvelope::Typed(env));
+                return Ok(());
+            }
+        }
+        // Try again
+        from.ctx().initialise_pool();
```

Comment:
Aehm...why are we unconditionally initialising the pool later here and then sending the message again?

Reply:
The `borrow_mut()` on line 316 remains in scope for the `from.ctx().initialise_pool()` call which causes a panic if I move it to an else in the same scope. 

The return on 326 prevents the initialise_pool from being reached unless I've severely missunderstood something? I suppose I could do the borrow mut outside the if and then explicitly drop it in the else section before initialising.

Reply:
Ah, you are right, of course. But it would be a lot more readable if the initialisation happened in an else branch. This return-and-jump logic is not very maintainable, I think. (Although, I admit I'm sometimes tempted into this myself^^)

Reply:
I absolutely agree, I would prefer the if else structure but I honestly have no idea how to achieve it in the function. 

I think `If let` and the scoping etc. it is still being figured out in rust so it's not worth spending time figuring out a neater way to write the function now imo.

Reply:
Ok, then at least make it clear in the comments that this is the `else`-branch of the `if let`-above, to avoid future confusion.